x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -148,6 +187,24 @@ public Automaton allowedActionsMatcher(String index) {
         return automatonList.isEmpty() ? Automatons.EMPTY : Automatons.unionAndMinimize(automatonList);
     }
 
+    public boolean testIndexMatch(String checkIndex, String checkPrivilegeName) {
```

Comment:
Please add javadocs explaining what this is doing including why we cache groups to automatons